torch/library.py

```diff
@@ -154,6 +154,48 @@ def _register_fake(self, op_name, fn, _stacklevel=1):
         handle = entry.abstract_impl.register(func_to_register, source)
         self._registration_handles.append(handle)
 
+    def _impl_with_aoti_compile(self, op_name, dispatch_key=''):
+        r'''Register the operator to use the AOTI-compiled implementation.
+
+        Args:
+            op_name: operator name (along with the overload) or OpOverload object.
+            dispatch_key: dispatch key that the input function should be registered for. By default, it uses
+                          the dispatch key that the library was created with.
+
+        Example::
+            >>> my_lib = Library("aten", "IMPL")
+            >>> my_lib._impl_with_aoti_compile("div.Tensor", "CPU")
+        '''
+        if dispatch_key == '':
+            dispatch_key = self.dispatch_key
+        assert dispatch_key in ['CPU', 'CUDA', 'XPU']
```

Comment:
our check should probably just be that the dispatch key contains DispatchKey::Dense (so that it can support all devices, not just these three)